cmd/fs-v1.go

```diff
@@ -473,6 +473,12 @@ func (fs fsObjects) CopyObject(srcBucket, srcObject, dstBucket, dstObject string
 // startOffset indicates the starting read location of the object.
 // length indicates the total length of the object.
 func (fs fsObjects) GetObject(bucket, object string, offset int64, length int64, writer io.Writer) (err error) {
+	// This is a special case with object whose name ends with
+	// a slash separator, we always return object not found here.
+	if hasSuffix(object, slashSeparator) {
```

Comment:
> CheckGetObjArgs would return error.

yeah, I know. 

By the way, checkGetObjArgs calls IsValidObjectName(), this latter is called only from two places. I think it is safe to modify the behavior of IsValidObjectName() and make it accept trailing slash and then we can move your code after checkGetObjArgs. The reason is that any invalid object name with a trailing slash will return object not found, it would be good to take care of that use case.